scripts/migrate-users.py

```diff
@@ -0,0 +1,237 @@
+model = manager.model()
+
+CODALAB_HOME = manager.codalab_home
+
+###############################################################
+# Load user data from Django table
+###############################################################
+django_config = read_json_or_die(os.path.join(CODALAB_HOME,
+                                              'website-config.json'))
+
+# Use default settings as defined in codalab-worksheets
+if 'database' not in django_config:
+    django_config['database'] = {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': 'codalab.sqlite3',
+    }
+
+# Create database connections
+if django_config['database']['ENGINE'] == 'django.db.backends.mysql':
+    import MySQLdb
+    from MySQLdb.cursors import DictCursor
+    db_params = {
+        'db': django_config['database']['NAME'],
+        'user': django_config['database']['USER'],
+        'passwd': django_config['database']['PASSWORD'],
+        'cursorclass': DictCursor,
+    }
+    if django_config['database'].get('HOST', None):
+        db_params['host'] = django_config['database']['HOST']
+    if django_config['database'].get('PORT', None):
+        db_params['port'] = django_config['database']['PORT']
+    django_db = MySQLdb.connect(**db_params)
+    django_cursor = django_db.cursor()
+elif django_config['database']['ENGINE'] == 'django.db.backends.sqlite3':
+    import sqlite3
+
+    # Use basic dict factory, not sqlite3.Row, for mutability
+    def dict_factory(cursor, row):
+        d = {}
+        for idx, col in enumerate(cursor.description):
+            d[col[0]] = row[idx]
+        return d
+
+    # Default location: ../codalab-worksheets/codalab/$NAME
+    sqlite_path = os.path.join(os.path.dirname(os.getcwd()),
+                               'codalab-worksheets',
+                               'codalab',
+                               django_config['database']['NAME'])
+    django_db = sqlite3.connect(sqlite_path)
+    django_db.row_factory = dict_factory
+    django_cursor = django_db.cursor()
+else:
+    raise UsageError("Invalid database engine %r" %
+                     django_config['database']['ENGINE'])
+
+# General SQL query should work on both MySQL and sqlite3
+django_cursor.execute("""
+    SELECT
+    cluser.id,
+    cluser.password,
+    cluser.last_login,
+    cluser.username AS user_name,
+    cluser.first_name,
+    cluser.last_name,
+    cluser.email,
+    cluser.date_joined,
+    cluser.organization_or_affiliation AS affiliation,
+    cluser.is_superuser,
+    email.verified AS is_verified
+    FROM authenz_cluser AS cluser
+    LEFT OUTER JOIN account_emailaddress AS email
+    ON cluser.id = email.user_id
+    WHERE cluser.id != -1 AND cluser.is_active = 1""")
+django_users = list(django_cursor.fetchall())
+
+###############################################################
+# Preprocess user data
+###############################################################
+
+# Generate new user ids using uuid
+for user in django_users:
```

Comment:
Keep the user IDs exactly the same. This user ID is used by the bundle service for it's permissions. If you change it, then all users will lose access to all their data.